### PR TITLE
Add Gain Module + Add Drive Modules

### LINKED ISSUE #7: Add more drive modules
Work in progress:

- Saturation module
- Overdrive module
- Distortion module

### PR DESCRIPTION
**Pull request by qtipee**
Close #7 

A new Gain module is available and can increase/decrease an incoming signal amplitude between a given interval. Note that the Gain module uses a linear scale to modify the signal's amplitude.

3 Drive modules are available. Each module uses the new Gain module to modify the incoming signal before applying the Drive effect:

- Saturation: soft and warm drive effect that uses tanh function to softly clip the incoming signal near -1.0 and 1.0.
- Overdrive: can be more aggressive than the saturation because the user can choose the threshold value at which values will start to clip. The user can also define the asymmetry of the Overdrive that can cause negative values to be more or less affected by the effect.
- Distortion: really aggressive drive effect that brutally clips the values above -1.0 and 1.0 resulting in a square-like waveform.

### CODE REVIEW

File: pydalboard/modules/drive/distortion.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from typing import ClassVar
```
Comment (g-wizzy) [inline pydalboard/modules/drive/distortion.py]: This import is unused. I assume you copied it from other modules which I coded earlier, and forgot to remove the `ClassVar` import.

**Review by g-wizzy — CHANGES_REQUESTED** [review]
I like your approach to this. However, I'd like to keep the `Gain` out of the `OverdriveParameters`, `DistorsionParameters` etc. I think these parameter classes should be plain dataclasses with no logic inside. I think the `Gain` instance is better suited inside of the `Distorsion` class itself. Do you agree ?

**Comment by qtipee** [comment]
> I like your approach to this. However, I'd like to keep the `Gain` out of the `OverdriveParameters`, `DistorsionParameters` etc. I think these parameter classes should be plain dataclasses with no logic inside. I think the `Gain` instance is better suited inside of the `Distorsion` class itself. Do you agree ?

Yeah I guess you're right. It does not change the code a lot but keep the Parameters clean so let's go for that!

File: pydalboard/modules/drive/distortion.py (hunk 1 above)
Comment (qtipee) [reply pydalboard/modules/drive/distortion.py]: Removed from the 3 drive modules.

**Review by g-wizzy — APPROVED** [review]

**Pull request merged**